src/io/flutter/run/FlutterConfigurationEditorForm.java

```diff
@@ -24,9 +29,25 @@
   private JTextField myAdditionalArgumentsField;
   private JTextField myBuildFlavorField;
   private JTextField myAttachArgsField;
+  private EnvironmentVariablesTextFieldWithBrowseButton myEnvironmentVariables;
+  private JLabel myEnvvarLabel;
+
+  private static final List<String> DESKTOP_PLATFORMS =
+    Lists.newArrayList("enable-linux-desktop", "enable-macos-desktop", "enable-windows-desktop", "enable-windows-uwp-desktop");
 
   public FlutterConfigurationEditorForm(final Project project) {
     initDartFileTextWithBrowse(project, myFileField);
+    final FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
+    if (sdk != null) {
+      final Set<String> platforms = sdk.queryConfiguredPlatforms(true);
+      for (String platform : platforms) {
+        if (DESKTOP_PLATFORMS.contains(platform)) {
```

Comment:
Why are environment variables disabled for desktop platforms?

Reply:
They are only enabled for desktop platforms. They are disabled (below) if we don't detect at least one desktop.